domain/src/main/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/fareattributes/FareAttribute.java

```diff
@@ -0,0 +1,266 @@
+         */
+        public FareAttributeBuilder transfers(final Integer transfers) {
+            this.transfers = Transfers.fromInt(transfers);
+            this.originalTransferInteger = transfers;
+            return this;
+        }
+
+        /**
+         * Sets field agencyId value and returns this
+         *
+         * @param agencyId identifies the relevant agency for a fare
+         * @return builder for future object creation
+         */
+        public FareAttributeBuilder agencyId(final @Nullable String agencyId) {
+            this.agencyId = agencyId; // TODO: to be modified see https://github.com/MobilityData/gtfs-validator/issues/109
+            return this;
+        }
+
+        /**
+         * Sets field transferDuration value and returns this
+         *
+         * @param transferDuration length of time in seconds before a transfer expires
+         * @return builder for future object creation
+         */
+        public FareAttributeBuilder transferDuration(final int transferDuration) {
+            this.transferDuration = transferDuration;
+            return this;
+        }
+
+        /**
+         * This methods returns an {@code EntityBuildResult} representing a row from fare_attributes.txt if the
+         * requirements from the official GTFS specification are met. Otherwise, method returns a collection of notices
+         * specifying the issues.
+         *
+         * @return {@link EntityBuildResult} representing a row from fare_attributes.txt if the requirements from the
+         * official GTFS specification are met. Otherwise, method returns a collection pf notices specifying the issues.
+         */
+        public EntityBuildResult<?> build() {
+            noticeCollection.clear();
+
+            if (price == null || price < 0 || fareId == null || currencyType == null ||
+                    !PaymentMethod.isEnumValueValid(originalPaymentMethodInteger) ||
+                    !Transfers.isEnumValueValid(originalTransferInteger) ||
+                    (transferDuration != null && transferDuration < 0)) {
+
+                if (price == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("fare_attributes.txt",
+                            "price", fareId));
+                } else if (price < 0) {
+                    noticeCollection.add(new FloatFieldValueOutOfRangeNotice("fare_attributes.txt",
+                            "price", fareId, 0, Float.MAX_VALUE, price));
+                }
+                if (fareId == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("fare_attributes.txt",
+                            "fare_id", fareId));
+                }
+                if (currencyType == null) {
+                    noticeCollection.add(new MissingRequiredValueNotice("fare_attributes.txt",
+                            "currency_type", fareId));
+                }
+                if (paymentMethod == null) {
```

Comment:
I figured .fromInt of PaymentMethod returns null when passed an invalid value. I leave my original comment as follow and suggest the following
```suggestion
                if (paymentMethod == null) { // True when PaymentMethod.isEnumValueValid(originalPaymentMethodInteger) returns false
```

ORIGINAL COMMENT
Would it be possible for `!PaymentMethod.isEnumValueValid(originalPaymentMethodInteger)` to be true while `if (paymentMethod == null)` is NOT?

If that is not the case, some comment explaining how it comes to be would be good. It's easy to understand the code flow checking for a `null` value before doing an if on it, less so to understand the flow when `isEnumValid` returns false

Reply:
Indeed, that part is not clear. Thanks for the suggestion, will add comments to clarify the code.